src/contract/assignments.rs

```diff
@@ -150,6 +151,25 @@ impl ToMerkleSource for OwnedRightsInner {
     }
 }
 
+impl MergeRevealed for OwnedRights {
+    fn merge_revealed(&mut self, other: &Self) -> Result<bool, String> {
+        if OwnedRightsInner(self.clone())
+            .to_merkle_source()
+            .commit_serialize()
+            != OwnedRightsInner(other.clone())
+                .to_merkle_source()
+                .commit_serialize()
+        {
+            Err(s!("Owned Right structure missmatch in merge operation"))
```

Comment:
Pls no stringly errors (unless you need to format in other string data), they must be typed

Reply:
Updated with dedicated error type.